ui/page/staking/overview.go

```diff
@@ -79,16 +89,33 @@ func (pg *Page) ID() string {
 // the page is displayed.
 // Part of the load.Page interface.
 func (pg *Page) OnNavigatedTo() {
-
 	pg.loadPageData() // starts go routines to refresh the display which is just about to be displayed, ok?
 
-	go pg.WL.GetVSPList()
-	// TODO: automatic ticket purchase functionality
-	pg.autoPurchaseEnabled.SetEnabled(false)
+	pg.updateTBToggle()
+}
+
+func (pg *Page) updateTBToggle() {
+	if pg.WL.MultiWallet.TicketBuyerConfigIsSet() {
+		_, walID, _, _ := pg.WL.MultiWallet.GetAutoTicketsBuyerConfig()
+		pg.autoPurchase.SetChecked(pg.WL.MultiWallet.IsAutoTicketsPurchaseActive(walID))
+	} else {
+		pg.autoPurchase.SetChecked(false)
+	}
 }
 
 func (pg *Page) loadPageData() {
-	pg.ticketPrice = dcrutil.Amount(pg.WL.TicketPrice()).String()
+	go func() {
+		ticketPrice, err := pg.WL.MultiWallet.TicketPrice()
+		if err != nil {
+			pg.ticketPrice = "0"
+		} else {
+			pg.ticketPrice = dcrutil.Amount(ticketPrice.TicketPrice).String()
+		}
+	}()
+
+	if len(pg.WL.MultiWallet.VspList) == 0 {
+		go pg.WL.MultiWallet.GetVSPList(pg.WL.Wallet.Net)
```

Comment:
If this data is used for display purposes, is the page refreshed once the data becomes available?

Reply:
yes it is refreshed.